src/telemetry/acrolinxInstrumentation.ts

```diff
@@ -0,0 +1,73 @@
+import { setupLogging } from './logs/logs-setup';
+import { Counters, createDefaultCounters, setupMetrics } from './metrics/metrics-setup';
+import { MeterProvider } from '@opentelemetry/sdk-metrics';
+import { Logger } from '@opentelemetry/api-logs';
+import { AccessToken } from 'src/common-types';
+import { AcrolinxEndpoint, IntService } from 'src';
+
+export class AcrolinxInstrumentation {
+  private static acrolinxInstrumentation: AcrolinxInstrumentation;
+  public instruments: Instruments | undefined = undefined;
+  private readonly config: TelemetryConfig;
+  private readonly intService: IntService;
+
+  private constructor(endpoint: AcrolinxEndpoint, config: TelemetryConfig) {
+    this.intService = new IntService(endpoint);
```

Comment:
Cannot we reused existing `IntService` instance here to avoid instantiation with new keyword in the constructor? (inversion of control)

Reply:
There is no existing IntService instance available in the SDK. 

 The IntService class is unnecessarily heavy holding entire AcrolinxEndpoint copy/ref inside it just to access two kind of utility functions `getJsonFromPath` and `postJsonToPath`. If we remove this functions outside of endpoint the IntService could become way lighter and could be accessed only with access_token. But that's for another PR.